dlt/extract/extractors.py

```diff
@@ -211,7 +211,11 @@ def _write_to_dynamic_table(self, resource: DltResource, items: TDataItems, meta
             if table_name in self._filtered_tables:
                 continue
             # allow to cache dynamic table hints if only table name is dynamic
-            if table_name not in self._table_contracts or resource._table_has_other_dynamic_hints:
```

Comment:
@rudolfix I had to revert his change that you made, because yielding multiple transformations with different schemas would not work without it. Relevant test is: `def test_multiple_transformations_in_function`. Maybe you should review this once again.

Reply:
right! but this prevents caching of the computed dynamic table hints which is 99% of cases we handle. my take:
convert `_table_has_other_dynamic_hints` into `has_other_dynamic_hints` which for transformation resources is always True and for others it falls back to the current behavior